tasks/main.yml

```diff
@@ -33,27 +33,28 @@
     state: directory
   when: postgres_exporter_user != 'postgres'
 
-- name: download
+- name: download & extract
   become: true
   become_user: root
-  get_url:
-    url: "{{ postgres_exporter_url }}"
-    checksum: "{{ postgres_exporter_checksum }}"
-    dest: "{{ prometheus_exporter_dist_dir }}/{{ postgres_exporter_signature }}"
-    owner: root
-    group: root
-    mode: "u=rwx,go=rx"
+  unarchive:
+    src: "{{ postgres_exporter_url }}"
+    dest: "{{ postgres_exporter_dist_dir }}"
+    owner: "{{ postgres_exporter_user }}"
+    group: "{{ postgres_exporter_group }}"
```

Comment:
The intent was for the *process* to run as postgres_exporter_user / group but for the _binaries_ to be owned by root:root. The configuration can be owned by root but must be readable by the postgres_exporter_group.

This way any compromise of the postgres_exporter process cannot be used to modify its configuration